controllers/controller.go

```diff
@@ -1186,7 +1190,11 @@ func newNginx(instanceMergedWithFlavors *v1alpha1.RpaasInstance, plan *v1alpha1.
 	}
 
 	replicas := instanceMergedWithFlavors.Spec.Replicas
-	if isAutoscaleEnabled(instanceMergedWithFlavors.Spec.Autoscale) {
+	if shutdown := instanceMergedWithFlavors.Spec.Shutdown; shutdown {
+		*replicas = 0
```

Comment:
in case of repicas is nil, may crash, you could replace to:
```
zero := 0
replicas = &zero
```

Reply:
or use the pointer lib: https://pkg.go.dev/k8s.io/utils@v0.0.0-20230220204549-a5ecb0141aa5/pointer